### PR TITLE
Make Hermite-like polynomials represent constant by (1,1,...1)

### PR DESCRIPTION
**Pull request by kronbichler**
Inspired by a comment regarding negative weights by @Rombur in #6014, I had the idea that we can use the one free weight to make the Hermite-like polynomials represent the constant function 1 by the coefficient values (1,1,...,1). This has turned out to also make the already good conditioning even better for most degrees.

To make sure that this property gets properly tested (it means that `P_1'(0) = -P_0'(0)`), I have added several more cases to the Hermite-like test.

With that change, I can use the `get_constant_modes` of the `FE_DGQ` base class rather than overloading it.

### CODE REVIEW

File: source/base/polynomial.cc [hunk 1]
```diff
@@ -1445,8 +1448,29 @@ namespace Polynomials
...
             this->lagrange_support_points[degree-1] = 1.;
 
-            // scale close to approximate maximum
-            this->lagrange_weight = 1./this->value(0.4*jacobi_roots(0));
+            // Select the weight to make the derivative of the sum of P_0 and
+            // P_1 in zero to be 0. The derivative in x=0 is simply given by
+            // p~(0)/auxiliary_zero+p~'(0) + a*p~(0), where p~(x) is the
+            // Lagrange polynomial in all points except the first one which is
+            // the same for P_0 and P_1, and a is the weight we seek here. If
+            // we solve this for a,. This ensures that the sum of all
```
Comment (Rombur) [inline source/base/polynomial.cc]: I think that you want to rewrite this sentence `If we solve this for a,.`

File: tests/base/polynomial_hermite_like.cc [hunk 2]
```diff
@@ -51,6 +51,20 @@ int main()
...
+  deallog << std::endl << "degree 3" << std::endl;
+  interpolation_conditions(HermiteLikeInterpolation::generate_complete_basis(3));
+  plot(HermiteLikeInterpolation::generate_complete_basis(3));
+  deallog << std::endl << "degree 4" << std::endl;
+  interpolation_conditions(HermiteLikeInterpolation::generate_complete_basis(4));
+  plot(HermiteLikeInterpolation::generate_complete_basis(4));
+  deallog << std::endl << "degree 6" << std::endl;
   interpolation_conditions(HermiteLikeInterpolation::generate_complete_basis(6));
   plot(HermiteLikeInterpolation::generate_complete_basis(6));
+  deallog << std::endl << "degree 6" << std::endl;
```
Comment (Rombur) [inline tests/base/polynomial_hermite_like.cc]: degree 9

**Comment by kronbichler** [comment]
Thanks for the comments @Rombur. I've fixed them now and also slightly extended the header file regarding the character of the basis (it is not a genuinely nodal one because of the derivative degrees of freedom, but nodal in the interior.

**Comment by kronbichler** [comment]
Also note that the original changelog from #5879 is still appropriate.

**Pull request merged**